src/app/editor.js

```diff
@@ -19,8 +19,9 @@ function Editor (loadingFromGist) {
   this.uploadFile = function (file, callback) {
     var fileReader = new FileReader();
 
-    SOL_CACHE_FILE = utils.fileKey(file.name);
+    var cacheName = utils.fileKey(file.name);
     fileReader.onload = function (e) {
+      SOL_CACHE_FILE = cacheName;
```

Comment:
The browser shouldn't be stalled while the file loads because the `FileReader` is asynchronous. The other thing shouldn't be a problem but I'll push up a change to be sure.